vendors/espressif/boards/esp32/aws_demos/config_files/aws_demo_config.h

```diff
@@ -62,4 +62,21 @@
 
 #define democonfigGREENGRASS_DISCOVERY_TASK_STACK_SIZE    ( configMINIMAL_STACK_SIZE * 12 )
 
+#define democonfigMEMORY_ANALYSIS
+
+#ifdef democonfigMEMORY_ANALYSIS
+    /* configTOTAL_HEAP_SIZE abstracted away to prevent compiler error." */
+    /* ESP32 heap size spans multiple memory locations. Please see ESP32 Heap Documentation to find total available heap size. */
+    #define democonfigMEMORY_ANALYSIS_HEAP_SIZE           0
+    #define democonfigSTACK_UNIT_TYPE                     "bytes"
+    #define democonfigMEMORY_ANALYSIS_STACK_DEPTH_TYPE    UBaseType_t
+    #define democonfigMEMORY_ANALYSIS_MIN_EVER_HEAP_SIZE()        xPortGetMinimumEverFreeHeapSize()
+    #if ( INCLUDE_uxTaskGetStackHighWaterMark == 1 )
+        #define democonfigMEMORY_ANALYSIS_STACK_WATERMARK( x )    uxTaskGetStackHighWaterMark( x )
+    #else
+        #define democonfigMEMORY_ANALYSIS_STACK_WATERMARK( x )    NULL
+    #endif /* if( INCLUDE_uxTaskGetStackHighWaterMark == 1 ) */
+#endif /* democonfigMEMORY_ANALYSIS */
```

Comment:
Are these specific to ESP? If not, should we place them in some common location?

Reply:
Most platforms should be the same, there has been some variance by platform.